pkg/scheduler/eventhandlers.go

```diff
@@ -171,6 +173,10 @@ func (sched *Scheduler) updatePodInSchedulingQueue(oldObj, newObj interface{}) {
 
 	logger.V(4).Info("Update event for unscheduled pod", "pod", klog.KObj(newPod))
 	sched.SchedulingQueue.Update(logger, oldPod, newPod)
+
```

Comment:
We shouldn't use `framework.UnscheduledPodUpdate`, instead we should split this update event into specific Pod events with `framework.PodSchedulingPropertiesChange(newPod, oldPod)`

Reply:
We got another potential problem here:

1. If QHint is disabled, and a plugin registered `PodUpdate` events.
2. Scheduler tried to schedule pod1 -> failed -> update pod1 status -> podUpdate is triggered -> all pods are back to activeQ

To prevent this, I added a check `queue.IsPodUpdated` in L180, so we can maintain behavior consistent with the scheduling queue.
But when should we remove this, I assume when Qhint become GAed?

Reply:
That's a nice catch. Can you add a comment on L180 to describe the problematic scenario that would happen if we didn't have `IsPodUpdated` there?